### DISCUSSION TITLE
How coroutines will work in V.

### DISCUSSION THREAD

**Discussion by Omnuchka** [vlang/v#15185] [General]
As I know in Golang, we can't control OS threads(only set this number) and Go control routines for us, but how it will work in Vlang? Can we will manually control threads and  cause coroutines work in it and how will be share memory, this information so interesting for me and can you create short list with main points of realization it.

**Comment by medvednikov** [comment]
I've been considering Go's model, but this is not set in stone yet.

This is the last big feature that still needs brainstorming and implementation.

**Comment by Omnuchka** [comment]
> I've been considering Go's model, but this is not set in stone yet.
> 
> This is the last big feature that still needs brainstorming and implementation.

Thanks, good will be see mind map for this solution.

**Comment by ylluminate** [comment]
Just to track, discussion thread: https://discordapp.com/channels/592103645835821068/592106336838352923/679739670321692891

**Comment by dumblob** [comment]
See the main GitHub thread about this topic: https://github.com/vlang/v/issues/1868 and also an overview of the topic: https://github.com/vlang/v/issues/3814 .

**Comment by ghost** [comment]
Go's model isn't appropriate for greater control

**Comment by dumblob** [comment]
@mobop for "greater control" there'll be `unsafe` primitives and intrinsics.

**Comment by MatejMagat305** [comment]
honesly, golang is great, but i do not know if you add coroutines like goroutines you probably will loose ability build OS and fast (zero overhead) call c api and cgo is realy anoying thinking to overhead, but this is only my opinion....

**Reply by JalonSolov** [reply]
V doesn't (and won't) have cgo - it already has direct C interoperability, no cgo nonsense required.

https://github.com/vlang/vinix shows you can write OS code with V.

**Reply by larpon** [reply]
Just because you need a runtime for scheduling of coroutines etc., does not mean you have to make the runtime mandatory, it can be switched off at compile time. You need to have the runtime enabled to use any builtin scheduling but an OS would just disable the runtime and schedule, whatever it needs to schedule, by itself.
